### PR TITLE
MAINT - Adds unit tests for CommonDataTypeVerifier and resolves some detected issues

### PR DESCRIPTION
**Pull request by coyotesqrl**
#### Checklist:
- [ ] SAML Spec Table of Contents documentation updated

### CODE REVIEW

File: ctk/common/src/test/kotlin/org/codice/compilance/verification/core/CommonDataTypeVerifierSpec.kt [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+            }
+        }
+
+        "bad Uri value fails" {
+            badUri(null)
+            badUri(null, SAMLCore_3_4_1_1_a)
+            badUri("")
+            badUri("   ")
+            badUri("/not/absolute")
+            badUri("/not/absolute", SAMLCore_3_4_1_1_a)
```
Comment (brjeter) [inline ctk/common/src/test/kotlin/org/codice/compilance/verification/core/CommonDataTypeVerifierSpec.kt]: ✏️ Could you use the `forAll` to test these combinations like you did above?

File: ctk/common/src/test/kotlin/org/codice/compilance/verification/core/CommonDataTypeVerifierSpec.kt [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+@Suppress("StringLiteralDuplication" /* Small duplicate test strings */)
+class CommonDataTypeVerifierSpec : StringSpec() {
+    init {
+        "null or blank string value fails" {
+            badString(null)
+            badString(null, SAMLCore_3_4_1_1_a)
+            badString("")
+            badString("   ")
+            badString("   ", SAMLCore_3_4_1_1_a)
```
Comment (brjeter) [inline ctk/common/src/test/kotlin/org/codice/compilance/verification/core/CommonDataTypeVerifierSpec.kt]: ✏️ Could you use the `forAll` to test these combinations like you did below?

File: ctk/common/src/test/kotlin/org/codice/compilance/verification/core/CommonDataTypeVerifierSpec.kt (hunk 1 above)
Comment (coyotesqrl) [reply ctk/common/src/test/kotlin/org/codice/compilance/verification/core/CommonDataTypeVerifierSpec.kt]: Good idea. 👍

**Comment by coyotesqrl** [comment]
build now

**Comment by blen-desta** [comment]
Hero'd successfully 👍 
Tests pass ✅

**Pull request merged**